modules/consentManagementGpp.js

```diff
@@ -0,0 +1,384 @@
+  }
+}
+
+/**
+ * Stores CMP data locally in module to make information available in adaptermanager.js for later in the auction
+ * @param {object} cmpConsentObject required; an object representing user's consent choices (can be undefined in certain use-cases for this function only)
+ */
+function storeConsentData(cmpConsentObject) {
+  consentData = {
+    gppString: (cmpConsentObject) ? cmpConsentObject.gppString : undefined,
+
+    fullGppData: (cmpConsentObject) || undefined,
+  };
+  consentData.applicableSections = (checkApplicableSectionsIsReady(cmpConsentObject)) ? cmpConsentObject.applicableSections
+    : (checkApplicableSectionIsReady(cmpConsentObject)) ? cmpConsentObject.applicableSection : [];
+  consentData.apiVersion = CMP_VERSION;
+  return consentData;
+}
+
+/**
+ * Simply resets the module's consentData variable back to undefined, mainly for testing purposes
+ */
+export function resetConsentData() {
+  consentData = undefined;
+  userCMP = undefined;
+  consentTimeout = undefined;
+  gppDataHandler.reset();
+}
+
+/**
+ * A configuration function that initializes some module variables, as well as add a hook into the requestBids function
+ * @param {{cmp:string, timeout:number, allowAuctionWithoutConsent:boolean, defaultGdprScope:boolean}} config required; consentManagement module config settings; cmp (string), timeout (int), allowAuctionWithoutConsent (boolean)
+ */
+export function setConsentConfig(config) {
+  config = config && config.gpp;
+  if (!config || typeof config !== 'object') {
+    logWarn('consentManagement.gpp config not defined, exiting consent manager module');
+    return;
+  }
+
+  if (isStr(config.cmpApi)) {
+    userCMP = config.cmpApi;
+  } else {
+    userCMP = DEFAULT_CMP;
+    logInfo(`consentManagement.gpp config did not specify cmp.  Using system default setting (${DEFAULT_CMP}).`);
+  }
+
+  if (isNumber(config.timeout)) {
+    consentTimeout = config.timeout;
+  } else {
+    consentTimeout = DEFAULT_CONSENT_TIMEOUT;
+    logInfo(`consentManagement.gpp config did not specify timeout.  Using system default setting (${DEFAULT_CONSENT_TIMEOUT}).`);
+  }
+
+  if (userCMP === 'static') {
+    if (isPlainObject(config.consentData)) {
+      staticConsentData = config.consentData;
+      consentTimeout = 0;
+    } else {
+      logError(`consentManagement.gpp config with cmpApi: 'static' did not specify consentData. No consents will be available to adapters.`);
+    }
+  }
+
+  logInfo('consentManagement.gpp module has been activated...');
+
+  if (!addedConsentHook) {
+    $$PREBID_GLOBAL$$.requestBids.before(requestBidsHook, 50);
+  }
+  addedConsentHook = true;
+  gppDataHandler.enable();
+  loadConsentData(); // immediately look up consent data to make it available without requiring an auction
+}
+config.getConfig('consentManagement', config => setConsentConfig(config.consentManagement));
+
+// TODO this function will likely change a bit once PR #9205 is merged to master
```

Comment:
@jsnellbaker 9205 has merged

Reply:
Thank you for heads-up, I'll start to work on the changes.

Reply:
I have pushed a commit that should include the proper changes.  I did some tests and it seems to work.  As a note, I tested combining both the GPP consent module and defining some gpp values in the `ortb2.regs` setConfig; the module's values did replace the ortb2 setConfig value's.  Those ortb2 setConfig values were carried over when the module was not active.